jekyll/_cci2/ecs-ecr.md

```diff
@@ -81,323 +81,57 @@ AWS_RESOURCE_NAME_PREFIX | Prefix for some required AWS resources. Should corres
+          repo: "${AWS_RESOURCE_NAME_PREFIX}"
+          region: ${AWS_DEFAULT_REGION}
+          tag: "${CIRCLE_SHA1}"
+      - ...
 ```
 
-**Note:**
-Recall that you must [use `BASH_ENV`](#set-environment-variables)
-to set interpolated environment variables.
-
-#### Push Image to ECR, Deploy, and Test Deployment
-
-[Push the image](https://docs.docker.com/engine/reference/commandline/push/) to ECR,
-then deploy the image
-and validate that the deployment succeeded. Refer to the complete [./deploy.sh](https://github.com/circleci/go-ecs-ecr/blob/master/deploy.sh) script for details.
+### Deploy the new Docker image to an existing AWS ECS service
+The `deploy-service-update` job of the aws-ecs orb creates a new task definition that is based on the current task definition, but with the new Docker image specified in the task definition's container definitions, and deploys the new task definition to the specified ECS service.
 
 ```yaml
-version: 2
-jobs:
-  # ...
-  deploy:
-    # ...
-    steps:
-      # ...
-      - run:
-          name: Push image to ECR
-          command: |
-            . venv/bin/activate
-            eval $(aws ecr get-login --region $AWS_DEFAULT_REGION --no-include-email)
-            docker push $AWS_ACCOUNT_ID.dkr.ecr.$AWS_DEFAULT_REGION.amazonaws.com/$ECR_REPOSITORY_NAME:$CIRCLE_SHA1
-      - run:
-          name: Deploy
-          command: |
-            . venv/bin/activate
-            export ECS_TASK_FAMILY_NAME="${AWS_RESOURCE_NAME_PREFIX}-service"
-            export ECS_CONTAINER_DEFINITION_NAME="${AWS_RESOURCE_NAME_PREFIX}-service"
-            export EXECUTION_ROLE_ARN="arn:aws:iam::$AWS_ACCOUNT_ID:role/${AWS_RESOURCE_NAME_PREFIX}-ecs-execution-role"
-            ./deploy.sh
-      - run:
-          name: Test deployment (Please manually tear down AWS resources after use, if desired)
-          command: |
-            . venv/bin/activate
-            TARGET_GROUP_ARN=$(aws ecs describe-services --cluster $ECS_CLUSTER_NAME --services $ECS_SERVICE_NAME | jq -r '.services[0].loadBalancers[0].targetGroupArn')
-            ELB_ARN=$(aws elbv2 describe-target-groups --target-group-arns $TARGET_GROUP_ARN | jq -r '.TargetGroups[0].LoadBalancerArns[0]')
-            ELB_DNS_NAME=$(aws elbv2 describe-load-balancers --load-balancer-arns $ELB_ARN | jq -r '.LoadBalancers[0].DNSName')
-            curl http://$ELB_DNS_NAME | grep "Hello World!"
+version: 2.1
+orbs:
+  aws-ecr: circleci/aws-ecr@0.0.2
+  aws-ecs: circleci/aws-ecs@0.0.3
+workflows:
+  build-and-deploy:
+    jobs:
+      - ...
+      - aws-ecs/deploy-service-update:
+          requires:
+            - aws-ecr/build_and_push_image
+          aws-region: ${AWS_DEFAULT_REGION}
+          family: "${AWS_RESOURCE_NAME_PREFIX}-service"
+          cluster-name: "${AWS_RESOURCE_NAME_PREFIX}-cluster"
+          container-image-name-updates: "container=${AWS_RESOURCE_NAME_PREFIX}-service,tag=${CIRCLE_SHA1}"
 ```
 
 ### Set Up a Workflow
```

Comment:
The config snippet here should be in sync with the actual config. I think this part could be removed or rephrased to explain that the `deploy-service-update` job depends on `build_and_push_image` because of the `requires` key.

Reply:
Fixed